src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/claimant/DefendantNocDashboardNotificationHandler.java

```diff
@@ -1,43 +1,92 @@
 import static uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.DashboardScenarios.SCENARIO_AAA6_DEFENDANT_NOC_CLAIMANT;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.DashboardScenarios.SCENARIO_AAA6_DEFENDANT_NOC_CLAIMANT_HEARING_FEE_TASK_LIST;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.dashboardnotifications.DashboardScenarios.SCENARIO_AAA6_DEFENDANT_NOC_CLAIMANT_TRIAL_ARRANGEMENTS_TASK_LIST;
 
 @Service
-public class DefendantNocDashboardNotificationHandler extends DashboardCallbackHandler {
+@RequiredArgsConstructor
+public class DefendantNocDashboardNotificationHandler extends CallbackHandler {
 
     private static final List<CaseEvent> EVENTS = List.of(CREATE_CLAIMANT_DASHBOARD_NOTIFICATION_FOR_DEFENDANT_NOC);
     public static final String TASK_ID = "CreateClaimantDashboardNotificationDefendantNoc";
 
-    public DefendantNocDashboardNotificationHandler(DashboardApiClient dashboardApiClient,
-                                                    DashboardNotificationsParamsMapper mapper,
-                                                    FeatureToggleService featureToggleService) {
-        super(dashboardApiClient, mapper, featureToggleService);
-    }
+    private final DashboardApiClient dashboardApiClient;
+    private final DashboardNotificationsParamsMapper mapper;
+    private final FeatureToggleService toggleService;
 
     @Override
     public String camundaActivityId(CallbackParams callbackParams) {
         return TASK_ID;
     }
 
     @Override
-    public List<CaseEvent> handledEvents() {
-        return EVENTS;
+    protected Map<String, Callback> callbacks() {
+        return Map.of(
+            callbackKey(ABOUT_TO_SUBMIT),
+            callbackParams -> toggleService.isLipVLipEnabled() ? configureScenarioForDefendantNoc(callbackParams) : emptyCallbackResponse(
+                callbackParams)
+        );
+    }
+
+    private CallbackResponse configureScenarioForDefendantNoc(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        ScenarioRequestParams params = ScenarioRequestParams.builder().params(mapper.mapCaseDataToParams(caseData)).build();
+
+        dashboardApiClient.recordScenario(
+            caseData.getCcdCaseReference().toString(),
+            SCENARIO_AAA6_DEFENDANT_NOC_CLAIMANT.getScenario(),
+            authToken,
+            params
+        );
+
+        if (isNull(caseData.getTrialReadyApplicant()) && SdoHelper.isFastTrack(caseData)) {
+            dashboardApiClient.recordScenario(
+                caseData.getCcdCaseReference().toString(),
+                SCENARIO_AAA6_DEFENDANT_NOC_CLAIMANT_TRIAL_ARRANGEMENTS_TASK_LIST.getScenario(),
+                authToken,
+                params
+            );
+        }
+
+        if (isNull(caseData.getHearingFeePaymentDetails())) {
```

Comment:
Is that a sufficient condition? Do we need to take into account potential failures?

Reply:
Good catch. I will add the payment failed condition. @m-meulendijks-v1

Reply:
You might also have to check if fee_payment_outcome is null in case of Help with Fees

Reply:
Sorry, just thought of this

Reply:
@m-meulendijks-v1 Changes done. Could you please have a look ?